celements-xwiki-core/src/main/java/com/celements/init/CelementsRequestFilter.java

```diff
@@ -64,9 +67,29 @@ public CelementsRequestFilter(
     this.xwikiProvider = xwikiProvider;
   }
 
+  public ExecutionContext preExecute(HttpServletRequest request,
+      HttpServletResponse response) throws WikiMissingException, ExecutionException,
+      ExecutionContextException, ServletContainerException {
+    return preExecute(getUrlAction(request.getRequestURI()), request, response);
+  }
+
+  private String getUrlAction(String requestPath) throws ExecutionException {
+    String[] urlParts = StringUtils.tokenizeToStringArray(requestPath, "/");
+    if ((urlParts.length > 2) && getRightService().validAction(urlParts[0])) {
+      return urlParts[0];
+    }
+    return "view";
+  }
+
+  private XWikiRightServiceImpl getRightService() throws ExecutionException {
+    return (XWikiRightServiceImpl) awaitWikiAvailability(XWikiConstant.MAIN_WIKI,
```

Comment:
better use `xwikiProvider.await` directly, no need to await main wiki updates to check actions.